hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -1548,7 +1548,14 @@ object PruneDeadFields {
         val cond2 = rebuildIR(cond, env, memo)
         val cnsq2 = rebuildIR(cnsq, env, memo)
         val alt2 = rebuildIR(alt, env, memo)
-        If.unify(cond2, cnsq2, alt2, unifyType = Some(requestedType))
+
+        if (cnsq2.typ == alt2.typ && requestedType == cnsq2.typ)
```

Comment:
ah, inlining this made it much clearer that it's not the right check now. Should be `if (cnsq2.typ.isOfType(alt2.typ))`

Reply:
Apologies, should have caught this, and a strategy I followed in subsequent PRs.